pkg/stanza/adapter/converter.go

```diff
@@ -148,20 +148,43 @@ func (c *Converter) workerLoop() {
 			}
 
 			resourceHashToIdx := make(map[uint64]int)
+			scopeIdxByResource := make(map[uint64]map[string]int)
 
 			pLogs := plog.NewLogs()
 			var sl plog.ScopeLogs
+
+			/**
+				[ {resourceId, scope name} ]
+			**/
 			for _, e := range entries {
 				resourceID := HashResource(e.Resource)
+				scopeName := e.ScopeName
```

Comment:
This doesn't add much for readability but does require an extra allocation. Can we leave it out and refer to the field directly?
```suggestion
```